library/src/scala/tasty/reflect/Printers.scala

```diff
@@ -646,6 +646,7 @@ trait Printers
                   n == "copy" ||
                   n.matches("copy\\$default\\$[1-9][0-9]*") || // default parameters for the copy method
                   n.matches("_[1-9][0-9]*") || // Getters from Product
+                  n == "writeReplace" || // Case class serialization
```

Comment:
That's not correct: the user is free to write their own writeReplace method (e.g. `Baz` in https://github.com/lampepfl/dotty/blob/bb3673cd3aec19cfe01494698d62c70aed496e20/tests/run/serialize.scala)

Reply:
It does not print the code only if the method is synthetic. The comment is misleading. I will update it.

Reply:
OK, we still need to properly print `classOf[X.type]` to decompile https://github.com/lampepfl/dotty/blob/91574bd6689518282437eee6f6a31bb287cf377d/tests/run/classof-object.scala :).

Reply:
Yes, we do but I want to merge the last PR from the decompiler project before doing any changes.